### ISSUE TITLE
How to train and test model

### ISSUE THREAD

**Issue by Minst2701** [LukasHedegaard/continual-skeletons#11]
Hello, when I use your code to train or test model again, I realize that setting forward mode by frame takes more time than forward mode by clip and I don't know when I should use forward mode frame or clip. Could you explain me in detail in this issue? how can I train the model exactly like yours?
To run model, I use: python models/cost_gcn_mod/cost_gcn_mod.py --id train_frame_mod_xview --gpus=1 --forward_mode frame --train --test --max_epochs=30 --optimization_metric loss --test --batch_size=16 --num_workers=8 --learning_rate=0.1 --weight_decay 0.00001 --dataset_normalization=0 --dataset_name ntu60 --dataset_classes /continual-skeletons/datasets/ntu60/classes.yaml --dataset_train_data /continual-skeletons/datasets/ntu60/xview/train_data_joint.npy --dataset_val_data /continual-skeletons/datasets/ntu60/xview/val_data_joint.npy --dataset_test_data /continual-skeletons/datasets/ntu60/xview/val_data_joint.npy --dataset_train_labels /continual-skeletons/datasets/ntu60/xview/train_label.pkl --dataset_val_labels /continual-skeletons/datasets/ntu60/xview/val_label.pkl --dataset_test_labels /continual-skeletons/datasets/ntu60/xview/val_label.pkl
I am looking forward to hearing from you!

**Comment by LukasHedegaard** [comment]
Hi @Minst2701 , I general I would recommend that you use the `forward` (i.e., forward by clip) for training on the spatio-temporal graphs provided by the datasets. The `forward_step` (i.e., forward by frame) can be used for fast inference when one frame is passed at a time.

**Comment by Minst2701** [comment]
@LukasHedegaard , I understand that I use forward_step (forward by frame) combining with pretrained weights to recognize actions real-time and use the forward (forward by clip) to evaluate models on available datasets. Is this right?

**Comment by LukasHedegaard** [comment]
Yes, that would be a good approach

**Comment by Minst2701** [comment]
Thank you for your useful answers very much!

**Comment by LukasHedegaard** [comment]
Hi @Minst2701 , do you have further questions or is it OK to close the issue?
